test/unit/sessions/job_entities/test_job_entities.py

```diff
@@ -280,6 +283,57 @@ def test_job_run_as_user(
             assert entity_obj.job_run_as_user.windows_settings.user == expected_user
             assert entity_obj.job_run_as_user.windows_settings.passwordArn == expected_password_arn
 
+    @pytest.mark.parametrize(
+        "job_run_as_user",
+        [
+            dict([key_val for key_val in (runAs, posix, windows) if key_val is not None])
+            for runAs in (("runAs", "QUEUE_CONFIGURED_USER"), ("runAs", "WORKER_AGENT_USER"))
+            for posix in (
+                (
+                    "posix",
+                    {
+                        "user": "job-user",
+                        "group": "job-group",
+                    },
+                ),
+                None,
+            )
+            for windows in (
+                (
+                    "windows",
+                    {
+                        "user": "job-user",
+                        "passwordArn": "job-password-arn",
+                    },
+                ),
+                None,
+            )
+        ],
+    )
+    def test_job_run_as_user_with_override(
+        self, job_run_as_user: dict, job_run_as_user_overrides: JobsRunAsUserOverride
+    ) -> None:
+        """Ensures that if we receive a job_run_as_user field in the response,
+        that the created entity has a SessionUser created with the
+        proper values"""
+        # GIVEN
+        api_response: dict = {
+            "jobId": "job-123",
+            "jobRunAsUser": job_run_as_user,
+            "logGroupName": "TEST",
+            "schemaVersion": TemplateSpecificationVersion.JOBTEMPLATE_v2023_09.value,
+        }
+
+        # WHEN
+        job_details_data = JobDetails.validate_entity_data(
+            api_response, job_user_override=job_run_as_user_overrides
+        )
+        entity_obj = JobDetails.from_boto(job_details_data)
```

Comment:
nit: shouldn't this test belong in `test/unit/sessions/job_entities/test_job_details.py` instaed, since we're testing behaviour of the `JobDetails` class?

Marked as a nit because I see this is not precedent as the test above has the same problem. 😞